source/common/http/codec_client.cc

```diff
@@ -121,18 +122,28 @@ void CodecClient::onReset(ActiveRequest& request, StreamResetReason reason) {
 
 void CodecClient::onData(Buffer::Instance& data) {
   bool protocol_error = false;
+  Status status;
   try {
-    codec_->dispatch(data);
+    status = codec_->dispatch(data);
+    // TODO(#10878): Remove this when exception removal is complete. It is currently in migration.
+    // Soon we won't need to catch these exceptions, as they'll be propagated through the error
+    // statuses callbacks and returned from dispatch.
```

Comment:
Yeah this is confusing. So the Codec::dispatch mechanism inconsistently returns statuses and exceptions right now? Might want to just point that out too.

Reply:
Now with option (2) from your factoring comment, dispatch now uniformly returns an error status which I like a lot better, but see that thread